### PR TITLE
Feat: adds huggingface bloom example to torchserve

### PR DESCRIPTION
**Pull request by jagadeeshi2i**
**What this PR does / why we need it**:

Adds Huggingface bloom example to Torchserve

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

**Review by chauhang — APPROVED** [review]

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+navigate to `examples/Huggingface_Transformers` and follow the steps for creating the MAR file including serialized model and other dependent files.
+TorchServe supports both eager model and torchscript and here we save as the pretrained model. 
+ 
+```bash
+torch-model-archiver --model-name BLOOMSeqClassification --version 1.0 \
+--serialized-file Transformer_model/pytorch_model.bin \
+--handler ./Transformer_handler_generalized.py \
+--extra-files "Transformer_model/config.json,./setup_config.json,./Seq_classification_artifacts/index_to_name.json"
+```
+## Create NVMe Persistent Volume
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md:23]: Can you explain what's `NVMe` and `lsblk` command?

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+## Model archive file creation
+
+Clone [pytorch/serve](https://github.com/pytorch/serve) repository,
+navigate to `examples/Huggingface_Transformers` and follow the steps for creating the MAR file including serialized model and other dependent files.
+TorchServe supports both eager model and torchscript and here we save as the pretrained model. 
+ 
+```bash
+torch-model-archiver --model-name BLOOMSeqClassification --version 1.0 \
+--serialized-file Transformer_model/pytorch_model.bin \
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]: how is this different from the other BERT model example?

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
+# TorchServe example with Huggingface BLOOM model
+In this example we will show how to serve [Huggingface Transformers with TorchServe](https://github.com/pytorch/serve/tree/master/examples/Huggingface_Transformers)
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]: Do you mean by `https://github.com/pytorch/serve/tree/master/examples/Huggingface_Largemodels` ?

File: docs/samples/v1beta1/torchserve/v2/README.md [hunk 4]
```diff
@@ -29,7 +29,7 @@ $inferenceservice.serving.kserve.io/torchserve-mnist-v2 created
 
 ## Inference with V2 REST Protocol
 
-The first step is to [determine the ingress IP and ports](https://kserve.github.io/website/master/get_started/first_isvc/#4-determine-the-ingress-ip-and-ports) and set `INGRESS_HOST` and `INGRESS_PORT`
+The first step is to [determine the ingress IP and ports](https://kserve.github.io/website/master/get_started/first_isvc/#4-determine-the-ingress-ip-and-portss) and set `INGRESS_HOST` and `INGRESS_PORT`
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v2/README.md]: why the change here?

File: docs/samples/v1beta1/torchserve/v1/bloom/storageclass.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+    name: fast-disks
+provisioner: kubernetes.io/no-provisioner
+volumeBindingMode: WaitForFirstConsumer
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/storageclass.yaml]: nit: newline

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM                     STORAGECLASS   REASON   AGE
+local-pv-2a85b8ac                          116Gi      RWO            Delete           Bound       kserve-test/model-cache   fast-disks              4d3h
+```
+## Create PVC and helper pod.
+
+```bash
+kubectl apply -f pvc-pod.yaml
+```
+
+## Move model to PVC
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]: This step can be merged to `Create PVC and Mount the model`

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+local-nvme-pv-provisioner-vwxgt              1/1     Running   0          16m
+```
+
+```bash
+$ kubectl get pv
+
+NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM                     STORAGECLASS   REASON   AGE
+local-pv-2a85b8ac                          116Gi      RWO            Delete           Bound       kserve-test/model-cache   fast-disks              4d3h
+```
+## Create PVC and helper pod.
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]:
```suggestion
## Create PVC and Mount the Model.
```

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md (hunk 6 above)
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]:
```suggestion
```

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+
+Expected Output
+
+```bash
+$inferenceservice.serving.kubeflow.org/torchserve-bloom-560m created
+```
+
+## Run a prediction
+
+The first step is to [determine the ingress IP and ports](../../../../../README.md#determine-the-ingress-ip-and-ports) and set `INGRESS_HOST` and `INGRESS_PORT`
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]:
```suggestion
The first step is to [determine the ingress IP and ports](https://kserve.github.io/website/latest/get_started/first_isvc/#4-determine-the-ingress-ip-and-ports) and set `INGRESS_HOST` and `INGRESS_PORT`
```

File: docs/samples/v1beta1/torchserve/v1/bloom/bloom-560m.yaml [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+spec:
+  predictor:
+    pytorch:
+      storageUri: pvc://model-cache
+      resources:
+          limits:
+            cpu: 4
+            memory: 16Gi
+            nvidia.com/gpu: 1
+  nodeName: ip-xxx-xxx-xxx-xxx.us-west-2.compute.internal
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/bloom-560m.yaml:14]: This does not look like valid syntax

File: docs/samples/v1beta1/torchserve/v2/README.md (hunk 4 above)
Comment (jagadeeshi2i) [reply docs/samples/v1beta1/torchserve/v2/README.md]: typo

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md (hunk 2 above)
Comment (jagadeeshi2i) [reply docs/samples/v1beta1/torchserve/v1/bloom/README.md]: This example covers, how to load and run inference for large HF models with Torchserve-KServe.

**Review by chauhang — APPROVED** [review]

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+Create a StorageClass 
+
+storageclass.yaml
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+name: fast-disks
+provisioner: kubernetes.io/no-provisioner
+volumeBindingMode: WaitForFirstConsumer
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]:
```suggestion
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: fast-disks
  provisioner: kubernetes.io/no-provisioner
  volumeBindingMode: WaitForFirstConsumer
```

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+Apply the CRD
+
+```bash
+kubectl apply -f bloom-560m.yaml
+```
+
+Expected Output
+
+```bash
+$inferenceservice.serving.kubeflow.org/torchserve-bloom-560m created
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]:
```suggestion
$inferenceservice.serving.kserve.io/torchserve-bloom-560m created
```

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+```
+
+## Run a prediction
+
+The first step is to [determine the ingress IP and ports](https://kserve.github.io/website/0.10/get_started/first_isvc/#4-determine-the-ingress-ip-and-ports) and set `INGRESS_HOST` and `INGRESS_PORT`
+
+```bash
+MODEL_NAME=torchserve-bert
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]: should be `torchserve-bloom-560m` ?

Comment (yuzisun) [reply docs/samples/v1beta1/torchserve/v1/bloom/README.md]: MODEL_NAME=BLOOMSeqClassification
SERVICE_NAME=torchserve-bloom-560m

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 13]
```diff
@@ -0,0 +1,197 @@
+# TorchServe example with Huggingface BLOOM model
+In this example we will show how to serve [Large Huggingface models with TorchServe](https://github.com/pytorch/serve/tree/master/examples/Huggingface_Largemodels)
+on KServe.
+
+## Model archive file creation
+
+Clone [pytorch/serve](https://github.com/pytorch/serve) repository,
+navigate to `examples/Huggingface_Largemodels` and follow the steps for creating the MAR file including serialized model and other dependent files.
+
+The above Torchserve example works on shard version on Huggingface models.
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]:
```suggestion
The above Torchserve example works on shard version of Huggingface models.
```

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+
+The For sharding Huggingface models use the below script, and continue with [this step](https://github.com/pytorch/serve/tree/master/examples/Huggingface_Largemodels#step-2-compress-downloaded-model)
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]:
```suggestion
For sharding the Huggingface models you can use the following script, and then [compress the model](https://github.com/pytorch/serve/tree/master/examples/Huggingface_Largemodels#step-2-compress-downloaded-model)
```

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+tokenizer = AutoTokenizer.from_pretrained(model_name)
+model.save_pretrained("model/"+model_name, max_shard_size="5GB")
+tokenizer.save_pretrained("model/"+model_name)
+```
+
+TorchServe supports both eager model and torchscript and here we save as the pretrained model. 
+ 
+```bash
+torch-model-archiver --model-name BLOOMSeqClassification --version 1.0 \
+--serialized-file Transformer_model/pytorch_model.bin \
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]: This is not the sharded model right?

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 16]
```diff
@@ -0,0 +1,197 @@
...
+--handler ./Transformer_handler_generalized.py \
+--extra-files "Transformer_model/config.json,./setup_config.json,./Seq_classification_artifacts/index_to_name.json"
+```
+
+## Create NVMe Persistent Volume
+
+Use SSH to connect to the worker nodes and prepare the NVMe drives for Kubernetes, as follows.
+
+Run the `lsblk`  command on each worker node to lists information about all available.
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]:
```suggestion
Run the `lsblk`  command on each worker node to list the available disks. 
```

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+
+Clone the local provisioner repository:
+
+```bash
+$ git clone https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner.git
+```
+
+Create a StorageClass 
+
+storageclass.yaml
```
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/README.md]:
```suggestion
Create a StorageClass yaml file `storageclass.yaml`
```

File: docs/samples/v1beta1/torchserve/v1/bloom/bloom-560m.yaml (hunk 9 above)
Comment (yuzisun) [inline docs/samples/v1beta1/torchserve/v1/bloom/bloom-560m.yaml:14]: Do we need this `nodeName` field?

File: docs/samples/v1beta1/torchserve/v1/bloom/README.md (hunk 15 above)
Comment (jagadeeshi2i) [reply docs/samples/v1beta1/torchserve/v1/bloom/README.md]: That is step is redundant. The Torchserve repo link cover MAR file generation step. I will remove it here.

File: docs/samples/v1beta1/torchserve/v1/bloom/bloom-560m.yaml (hunk 9 above)
Comment (jagadeeshi2i) [reply docs/samples/v1beta1/torchserve/v1/bloom/bloom-560m.yaml:14]: Yes, since we are provisioning storage from local disk. The isvc should be deployed in the node that has the volume.

**Comment by yuzisun** [comment]
Thanks @jagadeeshi2i !

/lgtm
/approve
